lib/matplotlib/__init__.py

```diff
@@ -105,6 +105,18 @@
 
 import six
 
+import sys
+if sys.version_info < (3, 5):  # noqa: E402
+    raise ImportError("""
+Matplotlib 3.0+ does not support Python 2.x, 3.0, 3.1, 3.2, 3.3, or 3.4.
+Beginning with Matplotlib 3.0, Python 3.5 and above is required.
```

Comment:
Both lines -> "Matplotlib 3.0+ requires Python 3.5 or above."

Reply:
This is the same as before and it's being explicit on purpose, as it's following IPython's lead.